### PR TITLE
Simplify the overriding of particular aspects of the Committed theme

### PR DESCRIPTION
**Pull request by jonnyelliot**
Introduces new optional props on ThemeProvider:

```
  createPalette?: (paletteColors: PaletteColors) => PaletteOptions
  createFonts?: () => typeof defaultFonts | undefined
  createShape?: () => ShapeOptions | undefined
  createSpacing?: () => SpacingOptions | undefined
  createTypography?: () =>
    | TypographyOptions
    | ((palette: Palette) => TypographyOptions)
    | undefined
  createOverrides?: (paletteColors: PaletteColors) => Overrides | undefined
```

If a prop is not specified, the default committed behaviour will occur. If the function returns undefined, the default material-ui behaviour will occur.

Removed the fonts prop. It is replaced by createFonts()

The default values of these function props are exported in case people want to merge or modify themes downstream. e.g. calling createTypography() but overriding the header font size.

### CODE REVIEW

**Comment by jonnyelliot** [comment]
Fixed secondary.primary palette.secondary they now use the correct colors. Extra handing has been added in createPalette() to maintain the previous behaviour when their are no custom palette colors.

**Comment by jonnyelliot** [comment]
This PR does not intend to make changes to stories. Not sure why this is a warning.

**Comment by stuarthendren** [comment]
If you change something then the sories (which are like the tests for this) would normally also change but that might not be the case, hense it's just a warning and not a fail.

Arguably, if you have added an important feature the stories should change to test, demonstrate and document the changes.

**Comment by jonnyelliot** [comment]
Fair point. I'll add to the theme provider story

File: src/theme/theme.ts [hunk 1]
```diff
@@ -77,19 +80,23 @@ export const defaultPaletteColors = {
   ...allColors,
   brand: allColors.committedYellow,
   primary: allColors.committedGrey,
-  secondary: allColors.committedYellow,
+  secondary: {
+    '300': allColors.committedYellow[50],
+    '500': allColors.committedYellow[200],
+    '700': allColors.committedYellow[400]
+  },
```
Comment (stuarthendren) [inline src/theme/theme.ts]: why is this just the three colours?

Comment (jonnyelliot) [reply src/theme/theme.ts]: it should do something like `darken(allColors.committedYellow, 0.5)`

**Comment by jonnyelliot** [comment]
Quite happy with the BC theme now

image

File: src/theme/ThemeProvider.tsx [hunk 2]
```diff
@@ -33,39 +46,58 @@ export interface ThemeProviderProps {
...
-    subheading: fonts.subheading || fonts.heading || defaultFont,
-    text: fonts.text || defaultFont,
-    display: fonts.display || defaultFont,
-    monospace: fonts.monospace || defaultFonts.monospace
-  }
-  const lightOptions = createLightOptions(
-    deepmerge(themes.defaultPaletteColors, paletteColors)
+  const paletteOptions = createPaletteOptions()
+  const palette = createMuiPalette(paletteOptions)
+  // createMuiPalette() "augments" inputted colors (than may be in several forms) to make them conform to {main: #xxxx, light:#xxxx ,...etc}
```
Comment (stuarthendren) [inline src/theme/ThemeProvider.tsx]:
```suggestion
  // createMuiPalette() "augments" inputted colors (that may be in several forms) to make them conform to {main: #xxxx, light:#xxxx ,...etc}
```

File: src/theme/themeProvider.stories.mdx [hunk 3]
```diff
@@ -55,6 +64,808 @@ Custom palette colours can be specified. For example, the primary, secondary or
...
+        </C.Box>
+      </C.Flex>
+    </Background>
+  </C.ThemeProvider>
+</Story>
+</Preview>
+
 ## Props
 
 <Props of={ThemeProviderDocs} />
```
Comment (stuarthendren) [inline src/theme/themeProvider.stories.mdx]: How do all the new possible props look here?
If not shown in this that may need documenting.

File: src/theme/theme.ts (hunk 1 above)
Comment (stuarthendren) [reply src/theme/theme.ts]: I assume this isn't needed?

File: src/theme/themeProvider.stories.mdx (hunk 3 above)
Comment (jonnyelliot) [reply src/theme/themeProvider.stories.mdx]: I've added all of the new props to the story now.

**Comment by stuarthendren** [comment]
I see you've added the other props to the example - but are they documented?
Sometimes you get the documentation for free as it can be inherited from material, but sometimes not.

If not I realise that it would be a lot to document but it this is essentially the same as what you can do in material then add a like to their docs. If there is a notable difference, just describe the difference.

**Comment by stuarthendren** [comment]
still needs the version bump too

**Comment by jonnyelliot** [comment]
I've added jsdoc for each prop in the ThemeProvider code.

I've also added a separate section for each prop in the story, describing what it does. I've linked off to material-ui where it makes sense to.

I've bumped the version to 2.0.0, due to breaking changes to fonts and color options.

**Comment by jonnyelliot** [comment]
> Warnings
> ⚠️	
> Changes were made to package.json, but not to yarn.lock - Perhaps you need to run `yarn install` or commit the lockfile?
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [254037c](https://github.com/commitd/components/commit/254037cfbb7603325768a08b0cde298f766a5383)

This is just due to the version change.

**Pull request merged**